components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1416,6 +1406,13 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         return this.teamDB.createTeam(user.id, name);
     }
 
+    public async joinTeam(teamId: string): Promise<void> {
```

Comment:
this is tricky. it basically allow me to join any team, just by knowing the id, thus the id serves as password. 
could a OTS help here? there is already a service in place.

Reply:
I don't think reusing OTS fits the purpose here well. We'll need to associate teams and invalidate the tokens on user action. For the email-based invite we'll also need to associate an email. I think we should introduce a dedicated table for this.